### ISSUE TITLE
Exit-node can't reach private network before private network subnets were advertised

### ISSUE THREAD

**Issue by W0n9** [tailscale/tailscale#9086]
### What is the issue?

After upgrading from 1.48.0 to 1.48.1, I found that I can't connect to my nas(192.168.123.118) by using exit-node function on my laptop(public network) via my server(192.168.123.24). And beyond that, I can't connect to my other VMs(located in 10.0.0.0/8 and 172.16.0.0/12) by using exit-node.
But I add an argument `--advertise-routes=192.168.123.118/32` on my server, the connection between my laptop and my nas resumed , although I didn't approve it on tailscale admin console. Under such circumstances, I can't connect to my home router(192.168.123.1), only my nas(192.168.123.118).
Later, I tried advertised all RFC1918 subsets on my server, all connections resumed.

### Steps to reproduce

ping node in private network from a device outside your private network which turn on exit-node function. 

### Are there any recent changes that introduced the issue?

update 1.48.0 to 1.48.1

### OS

Linux, Windows

### OS version

Windows 11, Ubuntu 22.04

### Tailscale version

1.48.1

### Other software

none

### Bug report

BUG-68006623a61094290b49cab92484b56ab22fb0d4aa8c0536823e7e11835d2a10-20230825035318Z-e88b6df62bdf8264

**Comment by kelivel** [comment]
@W0n9 Exit nodes don't allow access to its own LAN by default, and you'll need a subnet router to cover that range. Are you saying that you were able to access your exit node's LAN on older versions without having a subnet router set up? Were there other devices in the network advertising a subnet router that are no longer doing that?

**Comment by W0n9** [comment]
> @W0n9 Exit nodes don't allow access to its own LAN by default, and you'll need a subnet router to cover that range. Are you saying that you were able to access your exit node's LAN on older versions without having a subnet router set up? Were there other devices in the network advertising a subnet router that are no longer doing that?

Yes, I can access my exit node's LAN without a subnet router setup on older version.
Without any changes on advertising subnet router, just update tailscale version.
But I have found a workaround for this problem. Just advertising RFC1918 subsets on exit-node server without approve it.

**Comment by kelivel** [comment]
I went back to 1.46 and couldn't repro the behavior you originally reported, however I did reproduce the unapproved subnet router behavior in the latest client version. 

- iPhone [bugreport](http://go/bug/BUG-6915b5564173d34ced2c79c79685c98e25ebc3b138aaffb69a44f3994e10f2f7-20230928190842Z-d4dc9cacb4055ac0)
- Exit node [bugreport](http://go/bug/BUG-e25455aa0f1490111c88bfee4ae3ff7c3ba8785baa5a4ea249ea763cecce10a9-20230928191201Z-ef888f21862e25d6)

## Repro steps:
1. Disable approved subnet route (192.168.0.0/23) hosted on my exit node in the admin console.
2. Disable wifi on iPhone
3. Connect to Tailscale exit node
6. Ping 192.168.1.29, confirm ping succeeds **(it should not)**
7. Successfully browse to 192.168.1.118:5001 using my phone browser **(this should fail)**
8. Disable exit node
9. Confirm access to 192.168.1.118:5001 is revoked
4. Confirm ping to 192.168.1.29 fails

**Comment by kelivel** [comment]
@d-divita This sounds like a different problem from this GitHub issue. Can you open a ticket with our support team so we can take a closer look? http://www.tailscale.com/support — thanks!

**Comment by W0n9** [comment]
> I went back to 1.46 and couldn't repro the behavior you originally reported, however I did reproduce the unapproved subnet router behavior in 1.46 and the latest client version. This happens more specifically when the (deactivated) subnet router is hosted on the same machine as my exit node.
> 
> ![image](https://user-images.githubusercontent.com/104461838/271418839-904f3120-6b51-4d8c-8564-0be61a31f15c.png)
> 
> * iPhone [bugreport](http://go/bug/BUG-6915b5564173d34ced2c79c79685c98e25ebc3b138aaffb69a44f3994e10f2f7-20230928190842Z-d4dc9cacb4055ac0)
> * Exit node [bugreport](http://go/bug/BUG-e25455aa0f1490111c88bfee4ae3ff7c3ba8785baa5a4ea249ea763cecce10a9-20230928191201Z-ef888f21862e25d6)
> 
> ## Repro steps:
> 1. Deactivate approved subnet route (192.168.0.0/23) hosted on my exit node in the admin console.
> 2. Disable wifi on iPhone
> 3. Connect to Tailscale exit node
> 4. Ping 192.168.1.29, confirm ping succeeds **(it should not)**
> 5. Successfully browse to 192.168.1.118:5001 using my phone browser **(this should fail)**
> 6. Disable exit node
> 7. Confirm access to 192.168.1.118:5001 is revoked
> 8. Confirm ping to 192.168.1.29 fails

Yes, I think it is a server-side problem not a client-side problem.
Maybe the day when client upgradied to 1.48.1, there were something changes at server-side.

**Comment by pascalchristian** [comment]
> @W0n9 Exit nodes don't allow access to its own LAN by default, and you'll need a subnet router to cover that range. Are you saying that you were able to access your exit node's LAN on older versions without having a subnet router set up? Were there other devices in the network advertising a subnet router that are no longer doing that?

is this intended behaviour? it's not documented anywhere. this page https://tailscale.com/kb/1103/exit-nodes says that exit nodes means directing the default routing 0.0.0.0/0 through tailscale interface so it should allow access to the exit nodes' LAN without declaring additional subnet router.
